src/hakunapi-simple-servlet-javax/src/main/java/fi/nls/hakunapi/simple/servlet/javax/operation/GetCollectionItemByIdOperation.java

```diff
@@ -138,42 +145,51 @@ public Response handle(
-            GetFeatureRequest request, GetFeatureCollection c) throws Exception {
-        ValueProvider feature = null;
-        try (FeatureStream features = producer.getFeatures(request, c)) {
-            if (!features.hasNext()) {
-                return ResponseUtil.exception(Status.NOT_FOUND, "Feature not found");
+            GetFeatureRequest request, GetFeatureCollection c, RequestTelemetry ftt) throws Exception {
+
+        
+        try( TelemetrySpan span = ftt.span()) {
+
+            ValueProvider feature = null;
+            try (FeatureStream features = producer.getFeatures(request, c)) {
+                if (!features.hasNext()) {
+                    return ResponseUtil.exception(Status.NOT_FOUND, "Feature not found");
+                }
+                feature = features.next();
             }
-            feature = features.next();
-        }
 
-        int srid = request.getSRID();
+            int srid = request.getSRID();
 
-        int maxDecimalCoordinates = CrsUtil.getMaxDecimalCoordinates(srid);
-        boolean crsIsLatLon = service.isCrsLatLon(srid);
+            int maxDecimalCoordinates = CrsUtil.getMaxDecimalCoordinates(srid);
+            boolean crsIsLatLon = service.isCrsLatLon(srid);
 
-        // Feature response rarely needs 8kb of memory
-        // Let's allocate a little less
-        ByteArrayOutputStream baos = new ByteArrayOutputStream(2048);
-        writer.init(baos, maxDecimalCoordinates, srid, crsIsLatLon);
-        if (c.getFt().getGeom() != null) {
-            writer.initGeometryWriter(CrsUtil.getGeomDimensionForSrid(c.getFt().getGeomDimension(), srid));
-        }
 
-        int i = 0;
-        for (HakunaProperty prop : c.getProperties()) {
-            prop.write(feature, i++, writer);
-        }
+            // Feature response rarely needs 8kb of memory
+            // Let's allocate a little less
+            ByteArrayOutputStream baos = new ByteArrayOutputStream(2048);
+            writer.init(baos, maxDecimalCoordinates, srid, crsIsLatLon);
+            if (c.getFt().getGeom() != null) {
+                writer.initGeometryWriter(CrsUtil.getGeomDimensionForSrid(c.getFt().getGeomDimension(), srid));
+            }
+
+            int i = 0;
+            for (HakunaProperty prop : c.getProperties()) {
+                prop.write(feature, i++, writer);
+            }
 
-        writer.endFeature();
-        writer.end(true, getLinks(request, writer), 1);
-        writer.close();
+            writer.endFeature();
+            writer.end(true, getLinks(request, writer), 1);
+            writer.close();
 
-        ResponseBuilder builder = Response.ok();
-        request.getResponseHeaders().forEach((k, v) -> builder.header(k, v));
-        request.getFormat().getResponseHeaders(request).forEach((k, v) -> builder.header(k, v));
-        builder.entity(baos.toByteArray());
-        return builder.build();
+            span.counts(1);
```

Comment:
If telemetry fails then the whole operation fails. This seems more auditlog-gy (logging the operation is an essential part of the operation) - this is probably by design, just wanted to make sure.

Reply:
I think telemetry should not prevent operation.
Maybe we'll need a bunch of try catch-all additions to telemetry?

Reply:
In GetCollectionItemById it's probably simple as we prefetch the feature into memory and always call counts(1), so probably something like:
```
final Response response = builder.entity(baos.toByteArray()).build();
// Nothing can fail after this line, except the Telemetry, but we can catch it
try (TelemetrySpan span = ftt.span()) {
  span.counts(1);
} catch (Exception e) {
  // log or ignore the exception
}
return response;

Reply:
Spans have the ability to record the duration of operations.
We would lose that if we are not wrapping the actual operations with spans.

Maybe there should be a contract that "telemetry shall never fail"?

Reply:
In GetCollectionItemsOperation is probably a bit trickier due to streaming. I guess we could move `WriteReport report` to outside of the inner try block, e.g.:
```
WriteReport report = null;
try (FeatureStream features = producer.getFeatures(request, c);
                FeatureCollectionWriter writer = request.getFormat().getFeatureCollectionWriter()) {
    ...
    report = SimpleFeatureWriter.writeFeatureCollection(writer, ft, c.getProperties(), features, request.getOffset(), request.getLimit());
    ...
}
if (report != null) {
    try (TelemetrySpan span = ftt.span()) {
        span.counts(report);
    } catch (Exception e) {
        // Log or ignore
    }
}

Reply:
same comment as before - we would lose duration tracking with that change

Reply:
Oh okay. Yes then it would be best that telemetry implementations can never throw exceptions (checked or unchecked)

Reply:
kept this as is to have the possibility to trace spans and span durations